File: gpt-primes.go
package main

import "fmt"

func main() {
	fmt.Println("Enter the limit up to which prime numbers are to be found: ")
	var limit int
	fmt.Scanln(&limit)

	primes := make([]bool, limit+1)
	for i := 2; i <= limit; i++ {
		primes[i] = true
	}

	for p := 2; p*p <= limit; p++ {
		if primes[p] {
			for i := p * p; i <= limit; i += p {
				primes[i] = false
			}
		}
	}

	fmt.Println("Prime numbers up to", limit, "are:")
	for p := 2; p <= limit; p++ {
		if primes[p] {
			fmt.Println(p)
		}
	}
}

File: gpt-primes.py
#!/usr/bin/env python3

def is_prime(n):
    if n <= 1:
        return False
    for i in range(2, int(n**0.5)+1):
        if n % i == 0:
            return False
    return True

def generate_primes(n):
    primes = []
    for i in range(2, n+1):
        if is_prime(i):
            primes.append(i)
    return primes

n = int(input("Enter the value of n: "))
print("Prime numbers upto", n, "are:", generate_primes(n))

